src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -368,7 +474,8 @@ CHIP_ERROR ChipDnssdPublishService(const DnssdService * service, DnssdPublishCal
 
     auto regtype     = GetFullTypeWithSubTypes(service);
     auto interfaceId = GetInterfaceId(service->mInterface);
-    return Register(context, callback, interfaceId, regtype.c_str(), service->mName, service->mPort, record);
+
+    return Register(context, callback, interfaceId, regtype.c_str(), service->mName, service->mHostName, service->mPort, record);
```

Comment:
Do we really want to allow cases where the hostname is `null` or `empty` ? That seems do not enforce what this patch is trying to do :) 
I suggest to add assertions at the beginning of this method so we are only handling the case where there is an `hostname`.
If the hostname has not been found/resolved properly earlier I would like to caught those cases and fix them.
````
VerifyOrReturnError(service->mHostName != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
VerifyOrReturnError(strcmp(service->mHostName, "") != 0, CHIP_ERROR_INVALID_ARGUMENT);
````

Reply:
This was Boris's suggestion to use nullptr when hostname is null. i guess we can have a quick discussion when he's back. for now i will keep the fallback logic.

Reply:
done